src/lucas_kanade.jl

```diff
@@ -0,0 +1,234 @@
+
+The different arguments are:
+
+ -  prev_points       =  Vector of Coordinates for which the flow needs to be found
+ -  next_points       =  Vector of Coordinates containing initial estimates of new positions of
+                         input features in next image
+ -  window_size       =  Size of the search window at each pyramid level; the total size of the
+                         window used is 2*window_size + 1
+ -  max_level         =  0-based maximal pyramid level number; if set to 0, pyramids are not used
+                         (single level), if set to 1, two levels are used, and so on
+ -  estimate_flag     =  true -> Use next_points as initial estimate
+                         false -> Copy prev_points to next_points and use as estimate
+ -  term_condition    =  The termination criteria of the iterative search algorithm i.e the number of iterations
+ -  min_eigen_thresh  =  The algorithm calculates the minimum eigenvalue of a (2 x 2) normal matrix of optical
+                         flow equations, divided by number of pixels in a window; if this value is less than
+                         min_eigen_thresh, then a corresponding feature is filtered out and its flow is not processed
+                         (Default value is 10^-4)
+
+## References
+
+B. D. Lucas, & Kanade. "An Interative Image Registration Technique with an Application to Stereo Vision,"
+DARPA Image Understanding Workshop, pp 121-130, 1981.
+
+J.-Y. Bouguet, “Pyramidal implementation of the afﬁne lucas kanadefeature tracker description of the
+algorithm,” Intel Corporation, vol. 5,no. 1-10, p. 4, 2001.
+"""
+
+struct LK{T <: Int64, F <: Float64, V <: Bool} <: OpticalFlowAlgo
+    prev_points::Array{Coordinate{T}, 1}
+    next_points::Array{Coordinate{F}, 1}
+    window_size::T
+    max_level::T
+    estimate_flag::V
+    term_condition::T
+    min_eigen_thresh::F
+end
+
+LK(prev_points::Array{Coordinate{T}, 1}, next_points::Array{Coordinate{F}, 1}, window_size::T, max_level::T, estimate_flag::V, term_condition::T) where {T <: Int64, F <: Float64, V <: Bool} = LK{T, F, V}(prev_points, next_points, window_size, max_level, estimate_flag, term_condition, 0.0001)
+
+function optflow(first_img::AbstractArray{T, 2}, second_img::AbstractArray{T,2}, algo::LK{}) where T <: Gray
+    if algo.estimate_flag
+        @assert size(prev_points) == size(next_points)
+    end
+
+    # Construct gaussian pyramid for both the images
+    first_pyr = gaussian_pyramid(first_img, algo.max_level, 2, 1.0)
+    second_pyr = gaussian_pyramid(second_img, algo.max_level, 2, 1.0)
+
+    # Initilisation of the output arrays
+    guess_flow = Array{Coordinate{Float64},1}(size(algo.prev_points)[1])
+    final_flow = Array{Coordinate{Float64},1}(size(algo.prev_points)[1])
+    status_array = trues(size(algo.prev_points))
+    error_array = zeros(Float64, size(algo.prev_points))
+
+    for i = 1:size(algo.prev_points)[1]
```

Comment:
Since your data structure is an array of coordinates, you can just do
```julia
for i = 1:length(algo.prev_points)
```
instead of 
```julia
for i = 1:size(algo.prev_points)[1]
```
for your loops.

Reply:
Done